src/core/jpg.js

```diff
@@ -874,6 +874,13 @@ var JpegImage = (function JpegImageClosure() {
               offset -= 3;
               break;
             }
+            let nextFileMarker = findNextFileMarker(data, offset - 2);
+            if (nextFileMarker && nextFileMarker.invalid) {
+              warn('JpegImage.parse - unexpected image data, next marker is: ' +
+                   nextFileMarker.invalid);
```

Comment:
If I'm reading https://github.com/Snuffleupagus/pdf.js/blob/f2fed5d67181285a265b05f41e57d99fb2bda249/src/core/jpg.js#L631 correctly, `nextFileMarker.invalid` is actually the old marker (`currentMarker`), correct? Don't you want to print https://github.com/Snuffleupagus/pdf.js/blob/f2fed5d67181285a265b05f41e57d99fb2bda249/src/core/jpg.js#L632 (`newMarker`) here instead?

Reply:
I was simply following the same pattern as used elsewhere in the code, e.g.
https://github.com/mozilla/pdf.js/blob/29d77dedad84dafdfed04c8ca7d07b680c6cfce3/src/core/jpg.js#L369-L373
and
https://github.com/mozilla/pdf.js/blob/29d77dedad84dafdfed04c8ca7d07b680c6cfce3/src/core/jpg.js#L389-L393

For debugging purposes, I think it's much more interesting to know what the current *incorrect* marker is here (which was the purpose of these warnings).
However, your question made me realize that it's probably a poor choice of words here. So would replacing `next` with `current` here, and elsewhere in the file, perhaps be a better choice?

Reply:
Yes, I think that `unexpected data, current marker is:` would be better because I think that's what's being printed.

Reply:
I've made the changes at all the affected locations; thank you!

Re-reading https://github.com/mozilla/pdf.js/pull/9426#discussion_r165816331, I realize that I probably didn't answer all of your questions. If there something about `findNextFileMarker`, or its usage, that's unclear I'd be happy to try and explain!

Reply:
Looks better this way. Now that those warnings are changed, the term `invalid` makes sense to me again: `invalid` is the current invalid marker, and `marker`/`offset` are the new valid marker and its offset in the data stream.